src/store.rs

```diff
@@ -265,26 +271,50 @@ impl Store {
     }
 
     pub fn get(&self, id: &Scru128Id) -> Option<Frame> {
-        let res = self.partition.get(id.to_bytes()).unwrap();
+        let res = self.frame_partition.get(id.to_bytes()).unwrap();
         res.map(|value| serde_json::from_slice(&value).unwrap())
     }
 
-    pub fn head(&self, topic: String) -> Option<Frame> {
-        // Iterate over the partition in reverse order
-        let range: (Bound<Vec<u8>>, Bound<Vec<u8>>) = (Bound::Unbounded, Bound::Unbounded);
-        self.partition.range(range).rev().find_map(|record| {
-            let (_, value) = record.ok()?;
-            let frame: Frame = serde_json::from_slice(&value).ok()?;
-            if frame.topic == topic {
-                Some(frame)
-            } else {
-                None
-            }
-        })
+    pub fn head(&self, topic: &str) -> Option<Frame> {
+        let mut prefix = Vec::with_capacity(topic.len() + 1);
```

Comment:
In a future fjall version this will be possible to do without heap allocation (if the topic is short enough), because `fjall::Slice`s will have a constructor for fixed size with `Arc::get_mut`-like semantics.